### PR TITLE
core/txpool: use atomic int added in go1.19

### PR DESCRIPTION
**Pull request by s7v7nislands**
split the pr in https://github.com/ethereum/go-ethereum/pull/26820

### CODE REVIEW

File: core/txpool/txpool_test.go [hunk 1]
```diff
@@ -59,15 +59,21 @@ func init() {
...
 
 type testBlockChain struct {
-	gasLimit      uint64 // must be first field for 64 bit alignment (atomic access)
+	gasLimit      atomic.Uint64
 	statedb       *state.StateDB
 	chainHeadFeed *event.Feed
 }
 
+func newTestBlockChain(gasLimit uint64, statedb *state.StateDB, chainHeadFeed *event.Feed) *testBlockChain {
+	limit := atomic.Uint64{}
```
Comment (rjl493456442) [inline core/txpool/txpool_test.go:68]: nitpick, 

```go
var limit atomic.Uint64
limit.Store(gasLimit)
```

**Review by rjl493456442 — APPROVED** [review]
left a nitpick, otherwise lgtm

**Review by holiman — APPROVED** [review]
LGTM (but agree with @rjl493456442 's nitpick)

File: core/txpool/txpool_test.go (hunk 1 above)
Comment (s7v7nislands) [reply core/txpool/txpool_test.go]: updated

**Comment by holiman** [comment]
```
core/txpool/txpool_test.go:70:25: copylocks: literal copies lock value from limit: sync/atomic.Uint64 contains sync/atomic.noCopy (govet)
	return &testBlockChain{limit, statedb, new(event.Feed)}
	                       ^
util.go:48: exit status 1
exit
```

File: core/txpool/txpool_test.go [hunk 2]
```diff
@@ -59,15 +59,21 @@ func init() {
...
+	gasLimit      atomic.Uint64
 	statedb       *state.StateDB
 	chainHeadFeed *event.Feed
 }
 
+func newTestBlockChain(gasLimit uint64, statedb *state.StateDB, chainHeadFeed *event.Feed) *testBlockChain {
+	var limit atomic.Uint64
+	limit.Store(gasLimit)
+	return &testBlockChain{limit, statedb, new(event.Feed)}
+}
```
[the remark this answers is not in the document]
Comment (s7v7nislands) [reply core/txpool/txpool_test.go]: updated

**Pull request merged**